packages/common.vm/tools/vm.common/vm.common.psm1

```diff
@@ -1244,3 +1244,91 @@ public class Shell {
         VM-Write-Log-Exception $_
     }
 }
+
+
+# Usage example:
+# VM-Remove-DesktopFiles -excludeFolders "PS_Transcripts", ${Env:TOOL_LIST_DIR}, "fakenet_logs" -excludeFiles "example.txt", "important.doc"
+# The function is run against both the Current User and 'Public' desktops due to some cases where desktop icons showing on
+# Current user Desktop that are only located in Public/Desktop.
+function VM-Remove-DesktopFiles {
+    param (
+        [Parameter(Mandatory=$false)]
+        [string[]]$excludeFolders,
+        [Parameter(Mandatory=$false)]
+        [string[]]$excludeFiles
+    )
+    # Ensure that "PS_Transcripts", "Desktop\Tools", and "fakenet_logs" folders are not to be deleted.
+    $defaultExcludedFolders = @("PS_Transcripts", ${Env:TOOL_LIST_DIR}, "fakenet_logs")
+    $excludeFolders = $excludeFolders + $defaultExcludedFolders
+    $excludeFiles = $excludeFiles   # required for lint.ps1 to pass
```

Comment:
why is this exactly needed? What does the linter complain about?

Reply:
The linter says that the `$excludeFiles` variable isn't used, even though it is being used in the `if statement`. I think this might be a bug in how they are parsing things.

It said the same for `$excludeFolders` as well, but now that we add `+ $defaultExcludedFolders`, this is a little less obvious that I was doing the same thing for that variable as well.